pkg/yqlib/operator_filter.go

```diff
@@ -0,0 +1,38 @@
+package yqlib
+
+import (
+	"container/list"
+)
+
+func filterOperator(d *dataTreeNavigator, context Context, expressionNode *ExpressionNode) (Context, error) {
+	log.Warningf("-- filterOperation")
+	var results = list.New()
+
+	for el := context.MatchingNodes.Front(); el != nil; el = el.Next() {
+		candidate := el.Value.(*CandidateNode)
+		log.Warningf("candidate %#v", candidate)
+		children := context.SingleChildContext(candidate)
+		splatted, err := splat(children, traversePreferences{})
+		if err != nil {
+			return Context{}, err
+		}
+		filtered, err := selectOperator(d, splatted, expressionNode)
```

Comment:
OK I think I have this in a better spot now. I'm pushing out most of the logic to `select`, and it seems like its generating the proper arrays